google/cloud/spanner/samples/pg_samples.cc

```diff
@@ -0,0 +1,371 @@
+          .EmplaceRow(2, "Alice", "Bruxelles")
+          .Build();
+  auto insert_albums =
+      google::cloud::spanner::InsertOrUpdateMutationBuilder(
+          "Albums", {"SingerId", "AlbumId", "AlbumTitle", "MarketingBudget"})
+          .EmplaceRow(1, 1, "Total Junk", 100000)
+          .EmplaceRow(1, 2, "Go, Go, Go", 200000)
+          .EmplaceRow(2, 1, "Green", 300000)
+          .EmplaceRow(2, 2, "Forever Hold Your Peace", 400000)
+          .EmplaceRow(2, 3, "Terrified", 500000)
+          .Build();
+  auto commit = client.Commit(
+      google::cloud::spanner::Mutations{insert_singers, insert_albums});
+  if (!commit) throw std::runtime_error(commit.status().message());
+  std::cout << "Insert was successful.\n";
+}
+
+// [START spanner_postgresql_query_with_parameter]
+void QueryWithParameter(google::cloud::spanner::Client client) {
+  std::cout << "Listing all singers with a last name that starts with 'A'\n";
+  google::cloud::spanner::SqlStatement select(
+      "SELECT SingerId, FirstName, LastName FROM Singers"
+      " WHERE LastName LIKE $1",
+      {{"p1", google::cloud::spanner::Value("A%")}});
+  using RowType = std::tuple<std::int64_t, std::string, std::string>;
+  auto rows = client.ExecuteQuery(std::move(select));
+  for (auto const& row : google::cloud::spanner::StreamOf<RowType>(rows)) {
+    if (!row) throw std::runtime_error(row.status().message());
+    std::cout << "SingerId: " << std::get<0>(*row) << "\t";
+    std::cout << "FirstName: " << std::get<1>(*row) << "\t";
+    std::cout << "LastName: " << std::get<2>(*row) << "\n";
+  }
+  std::cout << "Query completed.\n";
+}
+// [END spanner_postgresql_query_with_parameter]
+
+// [START spanner_postgresql_dml_getting_started_update]
+void DmlGettingStartedUpdate(google::cloud::spanner::Client client) {
+  // A helper to read the budget for the given album and singer.
+  auto get_budget =
+      [&](google::cloud::spanner::Transaction txn, std::int64_t album_id,
+          std::int64_t singer_id) -> google::cloud::StatusOr<std::int64_t> {
+    auto key = google::cloud::spanner::KeySet().AddKey(
+        google::cloud::spanner::MakeKey(album_id, singer_id));
+    auto rows = client.Read(std::move(txn), "Albums", key, {"MarketingBudget"});
+    using RowType = std::tuple<absl::optional<std::int64_t>>;
+    auto row = google::cloud::spanner::GetSingularRow(
+        google::cloud::spanner::StreamOf<RowType>(rows));
+    if (!row) return std::move(row).status();
+    auto const budget = std::get<0>(*row);
+    return budget ? *budget : 0;
+  };
+
+  // A helper to update the budget for the given album and singer.
+  auto update_budget = [&](google::cloud::spanner::Transaction txn,
+                           std::int64_t album_id, std::int64_t singer_id,
+                           std::int64_t budget) {
+    auto sql = google::cloud::spanner::SqlStatement(
+        "UPDATE Albums SET MarketingBudget = $1"
+        "  WHERE SingerId = $3 AND AlbumId = $2",
```

Comment:
It doesn't look like there is a definitive source of what these samples are supposed to look like (just their names), but this one says it should be similar to the non-pg version, where these were ordered this way, although the non-pg versions uses names to bind instead of indices.  I could imagine that explicitly using out-of-order indices is a teaching point, but probably not worth it.  So, yes, swapped.